src/organizations/operations/update_organization.rs

```diff
@@ -0,0 +1,147 @@
+use std::collections::HashSet;
+
+use async_trait::async_trait;
+use reqwest::StatusCode;
+use serde::Serialize;
+use thiserror::Error;
+
+use crate::organizations::{Organization, OrganizationId, Organizations};
+use crate::{WorkOsError, WorkOsResult};
+
+/// The options for [`UpdateOrganization`].
+#[derive(Debug, Serialize)]
+pub struct UpdateOrganizationOptions<'a> {
+    /// The ID of the organization passed in the URL.
+    #[serde(skip_serializing)]
+    pub organization_id: &'a OrganizationId,
+
+    /// The name of the organization.
+    pub name: &'a str,
+
+    /// Whether the connections within this organization should allow profiles
+    /// that do not have a domain that is present in the set of the organization's
+    /// user email domains.
+    ///
+    /// See [here](https://workos.com/docs/sso/guide/frequently-asked-questions#allow-profiles-outside-organization)
+    /// for more details.
+    pub allow_profiles_outside_organization: Option<&'a bool>,
+
+    /// The domains of the organization.
+    ///
+    /// At least one domain is required unless
+    /// [`allow_profiles_outside_organization`] is `true`.
+    pub domains: HashSet<&'a str>,
```

Comment:
Let's make this field optional as well:

```suggestion
    pub domains: Option<HashSet<&'a str>>,
```